src/platform/flexpret.c

```diff
@@ -0,0 +1,115 @@
+#include "reactor-uc/platform/flexpret.h"
+
+static PlatformFlexpret platform;
+
+/**
+ * One variable per hardware thread so each thread knows whether an async
+ * event occurred
+ */
+static volatile bool _async_event_occurred[FP_THREADS] = THREAD_ARRAY_INITIALIZER(false);
```

Comment:
Hmm okay, now I'm a bit confused between the comment you left and what you're saying here. You say we need to handle "Programs where another FP thread is scheduling physical actions", but here we only support single-threaded operation. Can these two statements be true at the same time?

Reply:
I mean that currently we only support the reactor-uc runtime managing a single thread. However, we can have other FP threads running asynchronous to the reactor-uc runtime. E.g. running a networking stack, and scheduling events into the reactor-uc runtime through a physical actin

Reply:
While this is probably OK for a networking stack, it is exactly this pattern that I'm trying to avoid having LF programmers use.  Our goal should be that the LF programmer never explicitly starts a thread or acquires a mutex.